pkg/services/dashboards/service/integration_test/dashboard_k8s_test.go

```diff
@@ -0,0 +1,1680 @@
+package integration_test
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+
+	dashboardv1alpha1 "github.com/grafana/grafana/apps/dashboard/pkg/apis/dashboard/v1alpha1"
+	folderv0alpha1 "github.com/grafana/grafana/pkg/apis/folder/v0alpha1"
+	"github.com/grafana/grafana/pkg/services/dashboards/dashboardaccess"
+	"github.com/grafana/grafana/pkg/services/featuremgmt"
+	"github.com/grafana/grafana/pkg/services/folder"
+	"github.com/grafana/grafana/pkg/tests/apis"
+	"github.com/grafana/grafana/pkg/tests/testinfra"
+	"github.com/grafana/grafana/pkg/tests/testsuite"
+
+	"github.com/grafana/grafana/pkg/apimachinery/utils"
+)
+
+func TestMain(m *testing.M) {
+	testsuite.Run(m)
+}
+
+// TestContext holds common test resources
+type TestContext struct {
+	Helper                    *apis.K8sTestHelper
+	AdminUser                 apis.User
+	EditorUser                apis.User
+	ViewerUser                apis.User
+	TestFolder                *folder.Folder
+	AdminServiceAccountToken  string
+	EditorServiceAccountToken string
+	ViewerServiceAccountToken string
+	OrgID                     int64
+}
+
+// TestK8sDashboardIntegration tests the dashboard K8s API integration
+// These tests cover various scenarios including user types, permissions,
+// and validation for dashboard operations through the k8s API
+func TestK8sDashboardIntegration(t *testing.T) {
```

Comment:
```suggestion
func TestIntegrationK8sDashboard(t *testing.T) {
```

It'll have to start with `TestIntegration` to get picked up by ci

Reply:
TIL, thanks!